### DISCUSSION TITLE
<Image/> responsive. HOW?

### DISCUSSION THREAD

**Discussion by ilyasseisov** [vercel/next.js#69955] [Help]
### Summary

Hi 👋

I'm new to Next JS.

I'm coding a web app with Next JS and Tailwind CSS.

How do I make `<Image/>` responsive? What is the best practice?

```
 <Image
     alt="star"
     src={images.starLg}
     width={100}
     height={100}
     className="relative -top-4 ml-5 inline"
  />
```

Thanks in advance!

**Comment by Shriharsh-Deshmukh** [comment]
Hey **@ilyasseisov**  👋

To make `<Image />` responsive in Next.js, set the `layout` prop to `"responsive"` and adjust the `width` and `height` values to maintain the aspect ratio. Here's an example:

```jsx
<Image
  alt="star"
  src={images.starLg}
  width={100}
  height={100}
  layout="responsive"
  className="relative -top-4 ml-5 inline"
/>
```

This will automatically scale the image based on the parent container while keeping the aspect ratio. Tailwind will help with additional styling. Happy Coding! 😊

**Comment by ilyasseisov** [comment]
Thank you very much! 😊

On Wed, Sep 11, 2024 at 3:21 PM Shriharsh ***@***.***> wrote:

> Hey ***@***.*** <https://github.com/ilyasseisov>* 👋
>
> To make <Image /> responsive in Next.js, set the layout prop to
> "responsive" and adjust the width and height values to maintain the
> aspect ratio. Here's an example:
>
> <Image
>   alt="star"
>   src={images.starLg}
>   width={100}
>   height={100}
>   layout="responsive"
>   className="relative -top-4 ml-5 inline"/>
>
> This will automatically scale the image based on the parent container
> while keeping the aspect ratio. Tailwind will help with additional styling.
> Happy Coding! 😊
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/vercel/next.js/discussions/69955#discussioncomment-10612723>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQ5HP7NNEDPRHUTSKAHXENLZWAKSFAVCNFSM6AAAAABOAKRHQWVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTANRRGI3TEMY>
> .
> You are receiving this because you were mentioned.Message ID: <vercel/next
> .***@***.***>
>


-- 
<https://ilyasseisov.com/>

Ilyas Seisov

Front-End Web Developer
[image: globe]

ilyasseisov.com
[image: linkedin] <https://linkedin.com/in/ilyasseisov1> [image: twitter]
<https://twitter.com/ilyasseisov1>

**Comment by ilyasseisov** [comment]
Sorry)
Next JS says that "layout" is outdated prop )

On Thu, Sep 12, 2024 at 11:26 AM Ilyas Seisov ***@***.***> wrote:

> Thank you very much! 😊
>
> On Wed, Sep 11, 2024 at 3:21 PM Shriharsh ***@***.***>
> wrote:
>
>> Hey ***@***.*** <https://github.com/ilyasseisov>* 👋
>>
>> To make <Image /> responsive in Next.js, set the layout prop to
>> "responsive" and adjust the width and height values to maintain the
>> aspect ratio. Here's an example:
>>
>> <Image
>>   alt="star"
>>   src={images.starLg}
>>   width={100}
>>   height={100}
>>   layout="responsive"
>>   className="relative -top-4 ml-5 inline"/>
>>
>> This will automatically scale the image based on the parent container
>> while keeping the aspect ratio. Tailwind will help with additional styling.
>> Happy Coding! 😊
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/vercel/next.js/discussions/69955#discussioncomment-10612723>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AQ5HP7NNEDPRHUTSKAHXENLZWAKSFAVCNFSM6AAAAABOAKRHQWVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTANRRGI3TEMY>
>> .
>> You are receiving this because you were mentioned.Message ID:
>> ***@***.***>
>>
>
>
> --
> <https://ilyasseisov.com/>
>
> Ilyas Seisov
>
> Front-End Web Developer
> [image: globe]
>
> ilyasseisov.com
> [image: linkedin] <https://linkedin.com/in/ilyasseisov1> [image: twitter]
> <https://twitter.com/ilyasseisov1>
>
>
>

-- 
<https://ilyasseisov.com/>

Ilyas Seisov

Front-End Web Developer
[image: globe]

ilyasseisov.com
[image: linkedin] <https://linkedin.com/in/ilyasseisov1> [image: twitter]
<https://twitter.com/ilyasseisov1>
